src/Menu/Provider/GroupMenuProvider.php

```diff
@@ -170,4 +123,44 @@ public function has($name, array $options = [])
     {
         return 'sonata_group_menu' === $name;
     }
+
+    private function generateMenuItem(array $item, array $group)
+    {
+        if (isset($item['admin']) && !empty($item['admin'])) {
+            $admin = $this->pool->getInstance($item['admin']);
+
+            // skip menu item if no `list` url is available or user doesn't have the LIST access rights
+            if (!$admin->hasRoute('list') || !$admin->hasAccess('list')) {
+                return null;
```

Comment:
Consider avoiding `null` by either throwing or extracting this to a separate that will return a boolean.

Reply:
This will change the current behavior and be a possible BC-break. I am trying to keep the current behavior (but I can't use `continue` on a function.

Reply:
Throwing was IMO a bad idea since this is not really exceptional. See https://github.com/jordisala1991/SonataAdminBundle/pull/1 for another way to do this